File: app/libs/fusionAuthSDK_APP_Android_v1.2.12_operator_ui_log_static/Demo/demo/src/main/java/com/alicom/fusion/demo/MyApplication.java
package com.alicom.fusion.demo;

import android.app.Activity;
import android.app.Application;

import com.alicom.fusion.demo.utils.TokenActionFactory;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.CountDownLatch;

/**
 * @Package: com.alicom.fusion.demo
 * @Description:
 * @CreateDate: 2023/2/15
 */
public class MyApplication extends Application {

  private static  MyApplication instance;

  private List<Activity>activityList=new ArrayList<>();

    @Override
    public void onCreate() {
        super.onCreate();
        GlobalInfoManager.getInstance().setContext(this);

        initToken();
    }

    private void initToken(){
        new Thread(new Runnable() {
            @Override
            public void run() {
                CountDownLatch latch=new CountDownLatch(1);
                TokenActionFactory.getToken(GlobalInfoManager.getInstance().getContext());
                latch.countDown();
                try {
                    latch.await();
                } catch (InterruptedException e) {
                }
            }
        }).start();
    }



    public static MyApplication getInstance()

    {

        if(null == instance)

        {

            instance = new MyApplication();

        }

        return instance;

    }



    public void addActivity(Activity activity)

    {

        activityList.add(activity);

    }

//遍历所有Activity并finish

    public void exit()

    {

        for(Activity activity:activityList)

        {

            activity.finish();

        }

        System.exit(0);

    }
}

File: app/src/main/java/com/example/myapplication/MyApplication.kt
package com.example.myapplication

import android.app.Application
import android.content.Context
import com.example.myapplication.utils.ScreenAdaptUtil

/**
 * 应用程序入口类
 * 负责初始化屏幕适配等全局配置
 */
class MyApplication : Application() {
    
    companion object {
        private lateinit var instance: MyApplication
        
        fun getInstance(): MyApplication = instance
    }
    
    override fun onCreate() {
        super.onCreate()
        instance = this
        
        // 初始化屏幕适配
        initScreenAdapt()
    }
    
    /**
     * 初始化屏幕适配
     * 使用今日头条适配方案
     */
    private fun initScreenAdapt() {
        // 这里可以添加其他初始化逻辑
        // 屏幕适配会在Activity创建时调用
    }
    
    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        // 如果需要在这里进行屏幕适配，可以添加相关代码
    }
}
